sensor/kubernetes/certrefresh/tls_issuer_common.go

```diff
@@ -35,3 +49,167 @@ type certificateRefresherGetter func(certsDescription string, requestCertificate
+	}
+
+	certsRepo := i.getServiceCertificatesRepoFn(*sensorOwnerReference, i.sensorNamespace,
+		i.k8sClient.CoreV1().Secrets(i.sensorNamespace))
+	i.certRefresher = i.getCertificateRefresherFn(i.componentName, i.requestCertificates, certsRepo,
+		certRefreshTimeout, i.certRefreshBackoff)
+
+	if refreshStartErr := i.certRefresher.Start(); refreshStartErr != nil {
+		return i.abortStart(errors.Wrap(refreshStartErr, "starting certificate refresher"))
+	}
+
+	log.Debugf("%s TLS issuer started.", i.componentName)
+	return nil
+}
+
+func (i *tlsIssuerImpl) abortStart(err error) error {
+	log.Errorf("%s TLS issuer start aborted due to error: %s", i.componentName, err)
+	i.Stop(err)
+	return err
+}
+
+func (i *tlsIssuerImpl) Stop(_ error) {
+	i.stopSig.Signal()
+	if i.certRefresher != nil {
+		i.certRefresher.Stop()
+		i.certRefresher = nil
+	}
+
+	log.Debugf("%s TLS issuer stopped.", i.componentName)
+}
+
+func (i *tlsIssuerImpl) Notify(common.SensorComponentEvent) {}
+
+func (i *tlsIssuerImpl) Capabilities() []centralsensor.SensorCapability {
+	return []centralsensor.SensorCapability{i.sensorCapability}
+}
+
+// ResponsesC is called "responses" because for other SensorComponent it is Central that
+// initiates the interaction. However, here it is Sensor which sends a request to Central.
+func (i *tlsIssuerImpl) ResponsesC() <-chan *message.ExpiringMessage {
+	return i.msgToCentralC
+}
+
+// ProcessMessage dispatches Central's messages to Sensor received via the Central receiver.
+// This method must not block as it would prevent centralReceiverImpl from sending messages
+// to other SensorComponents.
+func (i *tlsIssuerImpl) ProcessMessage(msg *central.MsgToSensor) error {
+	response := i.getResponseFn(msg)
+	if response == nil {
+		// messages not supported by this component are ignored
+		return nil
+	}
+
+	go i.dispatch(response)
+	return nil
+}
+
+func (i *tlsIssuerImpl) dispatch(response *Response) {
+	if !i.requestOngoing.Load() {
+		log.Warnf("Received a response for request ID %q, but there is no ongoing RequestCertificates call.", response.RequestId)
+		return
+	}
+	if i.ongoingRequestID != response.RequestId {
+		log.Warnf("Request ID %q does not match ongoing request ID %q.",
+			response.RequestId, i.ongoingRequestID)
+		return
+	}
+
+	i.responseFromCentral = response
+	i.responseReceived.Signal()
+}
+
+// requestCertificates makes a new request for a new set of secured cluster certificates from Central.
+// Concurrent requests are *not* supported.
+func (i *tlsIssuerImpl) requestCertificates(ctx context.Context) (*Response, error) {
+	if i.requiredCentralCapability != nil {
+		// Central capabilities are only available after this component is created,
+		// which is why this check is done here
+		if !centralcaps.Has(*i.requiredCentralCapability) {
+			return nil, fmt.Errorf("TLS certificate refresh failed: missing Central capability '%s'", *i.requiredCentralCapability)
```

Comment:
I think, this branch is uncovered by test - in case you want to add the coverage.